Automattic/pocket-casts-android
# Add video clip sharing

## Description

This PR adds video clip sharing. Currently, the clip is always shared as a sheet due to the demo UI and only uses a vertical card.

I added the [Capturable](https://github.com/PatilShreyas/Capturable) library for rendering Compose UI to bitmaps. We already have code to render Compose UI to bitmap for EoY. However, from what I checked, it is from a tutorial made by the person who is the author of the library and now maintains it.

I decided to create the audio file first and then the video file. While this might be a bit inefficient, I found it to be more readable and safe to reuse the code we already have for audio creation since it deals with aspects like correct encoding and stream selection.

## Testing Instructions

> [!note]
> Video clip sharing for now works only with `debug` and `debugProd` variants.

1. Go to the clip creation.
2. Share a clip as video.
3. Wait until clip creation finishes and share it somewhere like Slack.
4. Play the video and check if it matches. Note that it might not be 100% accurate. This is something I plan to investigate: p1721029505573319/1721021230.484059-slack-C028JAG44VD

Run this scenario for different podcasts, a video podcast, and a downloaded file.

## Checklist
